### PR TITLE
chore: move `gnoland genesis` to `contribs/gnogenesis`

### LINKED ISSUE #2824: feedback needed: how are you using `gnoland genesis ...`?
I'm considering whether these commands should be included in the `gnoland` binary, placed elsewhere, or if they are necessary at all. 

I would appreciate feedback on how people are currently using these commands. If no one is using them yet, please explain how you envision using them in the future. 

Additionally, if you're already using them and have suggestions for improvement, please share your thoughts.

### PR DESCRIPTION
**Pull request by zivkovicmilos**
## Description

This PR migrates the `gnoland genesis` command suite under `contribs/gnogenesis`, after following discussions from #2824, and internal discussions.

**BREAKING CHANGE**
`gnoland genesis` will cease to exist after this PR, instead, you will need to use the binary in `contribs/gnogenesis`.
It's installed by default after running `make install` from the repo root.

Closes #2824 

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [x] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
</details>

### CODE REVIEW

**Comment by zivkovicmilos** [comment]
Heads up @sw360cab @r3v4s @sw360cab @albttx @D4ryl00

File: contribs/gnogenesis/go.mod [hunk 1]
```diff
@@ -0,0 +1,60 @@
+module github.com/gnolang/contribs/gnogenesis
+
+go 1.22
+
+require (
+	github.com/gnolang/gno v0.0.0-20241028125118-d03581ed748b
```
Comment (moul) [inline contribs/gnogenesis/go.mod:6]: Add a replace directive.

File: docs/gno-infrastructure/validators/faq.md [hunk 2]
```diff
@@ -107,7 +107,7 @@ either a full node or a pruned node, it is important to retain enough blocks to
 ### How do I generate `genesis.json`?
```
Comment (moul) [inline docs/gno-infrastructure/validators/faq.md:107]: This should be in gnogenesis/README.md, not in the validator's page.

**Review by moul — APPROVED** [review]
Thank you, Milos.

**Comment by leohhhn** [comment]
I'm not convinced this command should be installed by default with make install. It doesn't fit in the standard Gnome flow; like gno, gnodev, gnokey do. 

We should keep the happy dev flow clean and minimal imho.

File: docs/gno-infrastructure/validators/faq.md (hunk 2 above)
Comment (zivkovicmilos) [reply docs/gno-infrastructure/validators/faq.md]: I honestly believe this is sort of out of the scope of this PR, but I've added a fresh README regardless:

e667e9c

File: contribs/gnogenesis/go.mod (hunk 1 above)
Comment (zivkovicmilos) [reply contribs/gnogenesis/go.mod]: Added:

[e306256](https://github.com/gnolang/gno/pull/3041/commits/e306256aa2433d0c0ab3aea3c619028bea4fcfb5)

**Comment by zivkovicmilos** [comment]
@leohhhn 

>I don't think this command should be installed by default with make install. It doesn't fit in the standard Gnome dev flow like gno, gnodev, gnokey do.
>
>We should keep the happy dev flow clean and minimal imho.

You are more than welcome to open a PR that removes it from the "standard happy flow", and also improve tutorials on how this new binary can be used in development environments, if this is something in plan for the "gno developer roadmap".
I haven't seen much research done to support these ideas, given we have a very limited developer base to begin with, and most of them are advanced users of the `gnoland` binary.

I've kept it easy to install to avoid shell headaches. We can remove it definitely in the future.
The PR doesn't even introduce a Docker image, which is something we also need (cc @sw360cab)

**Pull request merged**